tests/qxmppuploadrequestmanager/tst_qxmppuploadrequestmanager.cpp

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (C) 2008-2019 The QXmpp developers
+ *
+ * Author:
+ *  Yury Gubich <blue@macaw.me>
+ *
+ * Source:
+ *  https://github.com/qxmpp-project/qxmpp
+ *
+ * This file is a part of QXmpp library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include <QObject>
+#include <QByteArray>
+#include <QMimeDatabase>
+#include "QXmppUploadRequestManager.h"
+#include "QXmppClient.h"
+#include "QXmppDiscoveryManager.h"
+#include "QXmppLogger.h"
+#include "util.h"
+
+class TestHelper : public QObject 
+{
+    Q_OBJECT
+public:
+    TestHelper(bool expectedEvent, bool expectedError);
+    ~TestHelper();
+    
+public slots:
+    void onSlotReceived(const QXmppHttpUploadSlotIq &slot);
+    void onRequestFailed(const QXmppHttpUploadRequestIq &request);
+    
+private:
+    bool expectedEvent;
+    bool expectedError;
+    bool event;
+    bool error;
+};
+
+class ManagerHeir : public QXmppUploadRequestManager {                  //i need this to give up protected methods to be able to test them
+public:
+    bool handleStanza(const QDomElement & stanza) override;
```

Comment:
I think it would be the best, if we just make `handleStanza()` public (other managers do the same).